### ISSUE TITLE
player crash on invalid track file.

### ISSUE THREAD

**Issue by Timus23** [florent37/Flutter-AssetsAudioPlayer#199]
**Flutter Version**

My version : 

**Lib Version**
1.17
My version : 

**Platform (Android / iOS / web) + version**

Platform : Android

**Describe the bug**

Player crashes when track file in invalid instead of giving some message. i hav seen the implementation here if exo player fails it tries to play from media player. if it also fails then their is no safety to prevent app from crashing. It would be better is we can prevent that bcz sometime track may b invalid due to server which may eventually crash mobile app.

**Comment by florent37** [comment]
can you find what's the methos that crash ? 

of is it directly inside exoplayer ?

**Comment by Timus23** [comment]
i think it indicate mediaplay.stop() in log. u can use https://d14nt81hc5bide.cloudfront.net/U7ZRzzHfk8pvmW28sziKKPzK

url it contain invalid track which will crash the app.

**Comment by florent37** [comment]
Ok I try

**Comment by Timus23** [comment]
i hav tried this same track in another player. it doesnot crash instead throw playerstate.error.
https://pub.dev/packages/music_player

i hav not seen implementation of this player but it might help to solve this problem.

**Comment by florent37** [comment]
I've opened your url in this code

```
try {
                          await _assetsAudioPlayer.open(myAudio,
                              autoStart: true,
                              respectSilentMode: true,
                              showNotification: true,
                              playInBackground: PlayInBackground.enabled,
                              notificationSettings: NotificationSettings(
                                //seekBarEnabled: false,
                                //stopEnabled: true,
                                //customStopAction: (player){
                                //  player.stop();
                                //}
                                //prevEnabled: false,
                                //customNextAction: (player) {
                                //  print("next");
                                //}
                              ));
                        } catch (e){
                          print(e);
                        }
```

and it does not crash but catch an exception inside the catch closure

**Comment by florent37** [comment]
oh ok I see it's when you click on play, it crash with platformexception

**Comment by Timus23** [comment]
> oh ok I see it's when you click on play, it crash with platformexception

yes. exactly. sry i forget to mention that.

**Comment by florent37** [comment]
ok I think i fixed that :)

**Comment by florent37** [comment]
published in `2.0.2+6`
